### ISSUE TITLE
Differentiating between native HA users and externally maintained ones

### ISSUE THREAD

**Issue by efficiosoft** [home-assistant/architecture#184]
Hi,

I want to start a discussion about classifying users in HA into two categories:

1. native users: those created and maintained directly in HA and authenticated via the homeassistant auth provider

2. external users: those created by an external auth provider like command_line

For the second category, the auth provider could be able to modify user information, such as name and group memberships upon authentication to keep them in sync with a central single signon database.

Some work, especially for partial user information updates, has been done in home-assistant/home-assistant#20550, but this PR just allows every auth provider to alter user information, which, although it doesn't create problems in practice, isn't desirable in case of multiple auth providers authenticating the same user.

How about a new per-user field in the user store to denote whether a user should be manageable by an auth provider or not, or maybe even a list of auth provider instances that are allowed to manage the user. The auth provider that created an user account could be added automatically upon creation, and the field could be made editable from the config UI later.

Best regards
Robert

**Comment by balloob** [comment]
I don't want to consider/discuss this right now. The auth system is still transitioning the last pieces of auth from `http/auth.py` to the correct auth providers. We are still making sure that context is passed around correctly and that permissions are enforced. I don't want to complicate it even further by introducing more advanced features that are not going to be used by the majority of Home Assistant users.

**Comment by efficiosoft** [comment]
@balloob Right, then I'm going to bring this up again in - let's say - about three months. Does that sound reasonable to you?

I'm still thinking that this is a feature which, when integrated cleanly, all users with a central auth database will benefit from a lot. These are for sure not the majority of users, especially not in small private environments, but given that external authentication is only possible since 0.88, the number will probably grow within the next months.

**Comment by frenck** [comment]
This architecture issue is old, stale, and possibly obsolete. Things changed a lot over the years. Additionally, we have been moving to discussions for these architectural discussions.

For that reason, I'm going to close this issue.

../Frenck
